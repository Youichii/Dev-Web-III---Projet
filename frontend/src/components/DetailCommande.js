import PropTypes from 'prop-types'

const DetailCommande = ({ informations, type_couleur, bg_bouton, onMouseOver, onMouseLeave, onClick_panier, onClick_ok }) => {
    return (
        <div className="i_commande c_commande" id={informations.IdClient} onMouseOver={onMouseOver} onMouseLeave={onMouseLeave}>
                <div className="c_sans_bouton i_sans_bouton" onClick={onClick_panier}>
                    <div className={`i_nom ${type_couleur}`}>{informations.Prenom}</div> 
                    <div className={`i_contact ${type_couleur}`}>{informations.Gsm}</div> 
                    <div className={`i_prix_commande ${type_couleur}`}>{informations.Prix}</div> 
                    <div className={`i_heure_prevue ${type_couleur}`}>{informations.HLivree}</div>
                </div>
                <div className={`i_div_bouton ${bg_bouton}`}>
                    <button className="i_bouton_suivant" onClick={onClick_ok}>OK</button>
                </div> 
            </div>
    )

}

/*DetailCommande.defaultProps = {
    color: 'steelblue',
    text : 'Hello'
}*/

DetailCommande.propTypes = {
    informations : PropTypes.string,
    type_couleur : PropTypes.string,
    bg_bouton : PropTypes.string,
    type_actuel : PropTypes.string,
    type_suivant : PropTypes.string,

    onMouseOver: PropTypes.func,
    onMouseLeave: PropTypes.func,
    onClick_panier: PropTypes.func,
    onClick_ok: PropTypes.func
}

export default DetailCommande
