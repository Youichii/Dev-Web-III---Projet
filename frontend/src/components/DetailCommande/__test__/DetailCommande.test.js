import React from 'react';
import ReactDOM from 'react-dom';
import DetailCommande from '../DetailCommande';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer  from "react-test-renderer";

//npm test to start all tests
afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<DetailCommande></DetailCommande>, div)
})

/*it("render button text correctly", () => {
    const {getByTestId} = render(<DetailCommande color="green" text='Hello' onClick={()=> {console.log("hello")}} ></DetailCommande>);
    expect(getByTestId('button')).toHaveTextContent("Hello")
})

it("render button color correctly", () => {
    const {getByTestId} = render(<DetailCommande color="green" text='Hello' onClick={()=> {console.log("hello")}} ></DetailCommande>);
    expect(getByTestId('button')).toHaveStyle("backgroundColor : green")
})

it("matches snapshot 1", () => {
    const tree = renderer.create(<DetailCommande text="save" color="blue"></DetailCommande>).toJSON(); //convert to a virtualDOM object
    expect(tree).toMatchSnapshot(); //create folder __snapshot__
}) // to update a Snapshot if it failed press -u in the terminal

it("matches snapshot 2", () => {
    const tree = renderer.create(<DetailCommande text="Hello" color="blue"></DetailCommande>).toJSON(); //convert to a virtualDOM object
    expect(tree).toMatchSnapshot();
}) // creating a 2nd snapshot*/