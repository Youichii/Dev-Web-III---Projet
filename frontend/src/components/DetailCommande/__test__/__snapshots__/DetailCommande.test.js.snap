// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 DetailCommande 1`] = `
<div
  className="i_commande c_commande"
  data-testid="div_detail_commande"
  id={7}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="c_sans_bouton i_sans_bouton"
    onClick={[Function]}
  >
    <div
      className="i_nom couleur_bg1"
    >
      Marie
    </div>
    <div
      className="i_contact couleur_bg1"
    >
      +32468571325
    </div>
    <div
      className="i_prix_commande couleur_bg1"
    >
      409.3
    </div>
    <div
      className="i_heure_prevue couleur_bg1"
    >
      18:00
    </div>
  </div>
  <div
    className="i_div_bouton var(--bg_ligne1)"
  >
    <button
      className="i_bouton_suivant"
      onClick={[Function]}
    >
      OK
    </button>
  </div>
</div>
`;

exports[`matches snapshot 2 DetailCommande 1`] = `
<div
  className="i_commande c_commande"
  data-testid="div_detail_commande"
  id={7}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="c_sans_bouton i_sans_bouton"
    onClick={[Function]}
  >
    <div
      className="i_nom couleur_bg1"
    >
      Marie
    </div>
    <div
      className="i_contact couleur_bg1"
    >
      +32468571325
    </div>
    <div
      className="i_prix_commande couleur_bg1"
    >
      409.3
    </div>
    <div
      className="i_heure_prevue couleur_bg1"
    >
      18:00
    </div>
  </div>
  <div
    className="i_div_bouton var(--bg_ligne1)"
  >
    <button
      className="i_bouton_suivant"
      onClick={[Function]}
    >
      OK
    </button>
  </div>
</div>
`;
